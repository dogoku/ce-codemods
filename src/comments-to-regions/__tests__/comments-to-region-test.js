'use strict';

const defineSnapshotTests = require('../../../util/defineSnapshotTests');

describe('comments-to-region', function() {
  defineSnapshotTests(__dirname);
});
