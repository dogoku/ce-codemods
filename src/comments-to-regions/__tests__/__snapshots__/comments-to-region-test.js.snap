// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments-to-region snapshot tests block-comments 1`] = `
"'use strict';

/**
 * A custom element
 *
 * @module my-element
 * @type {HTMLElement}
 */

function helper() {
  return 'help';
}

var dependency = require('a-dependency');

var MyElementProto = Object.create(HTMLElement.prototype);
MyElementProto.STA_TIC = 'static property';

var _inTheScope = 42;

//#region Custom Element API

MyElementProto.createdCallback = function() {
  this._data = null; /* Ignore me */
  this._onClick = this._onClick.bind(this);
  this.innerHTML = '<span>OH HAI</span>';
};

MyElementProto.attachedCallback = function() {
  this.addEventListener('click', this._onClick);
};

MyElementProto.detachedCallback = function() {
  this.removeEventListener('click', this._onClick);
  this._destroy();
};

//#endregion

//#region Public Methods

MyElementProto.getData = function() {
  return this._data;
};

MyElementProto.setData = function(data) {
  this._data = data;
};

//#endregion

//#region Private Methods

MyElementProto._onClick = function(event) {
  console.log('ORLY?')
};

//#endregion

//#region Registeration of the Custom Element

module.exports = document.registerElement('my-element', { prototype: MyElementProto });
module.exports.STA_TIC = MyElementProto.STA_TIC;

//#endregion"
`;

exports[`comments-to-region snapshot tests line-comments 1`] = `
"function helper() {
  return 'help';
}

//#region Custom Element API

MyElementProto.createdCallback = function() {
  this._data = null; // ignore me
  this._onClick = this._onClick.bind(this);
  this.innerHTML = '<span>OH HAI</span>';
};

//#endregion

//#region Private Methods

MyElementProto._onClick = function(event) {
  console.log('ORLY?')
};

//#endregion

//#region Registeration of the Custom Element

module.exports = document.registerElement('my-element', { prototype: MyElementProto });
module.exports.STA_TIC = MyElementProto.STA_TIC;

//#endregion"
`;
